### PR TITLE
#247 Add ktlint

### PR DESCRIPTION
**Pull request by nighthawk24**
Upgrade Gradle
Fix formatting for ktlint

### CODE REVIEW

**Comment by gmale** [comment]
Thanks @nighthawk24 for doing this! This is great. Since it touches 73 files, I'll get the other PRs merged first and then get this one in. I'll take care of any merge conflicts and keep you posted, if I run into any issues (I don't anticipate that I will).

**Review by gmale — COMMENTED** [review]
Thanks @nighthawk24 for doing this! This is great. Since it touches 73 files, I'll get the other PRs merged first and then get this one in. I'll take care of any merge conflicts and keep you posted, if I run into any issues (I don't anticipate that I will).

**Comment by nighthawk24** [comment]
@gmale can we merge ktlint checks before https://github.com/zcash/zcash-android-wallet/pull/253 and avoid resolving issues in possible ktlint fails

File: app/build.gradle [hunk 1]
```diff
@@ -12,7 +12,7 @@ group = 'cash.z.ecc.android'
 version = Deps.versionName
 
 android {
-    ndkVersion "21.1.6352462"
+    ndkVersion "21.4.7075529"
```
Comment (gmale) [inline app/build.gradle:15]: Do we need this change for ktlint? If not, can we skip it since it might be out of scope?

This moves the NDK version to one that does not match what the SDK is built with. The SDK's version should probably be updated in tandem with this.

File: app/src/main/java/cash/z/ecc/android/ui/send/SendAddressFragment.kt [hunk 2]
```diff
@@ -0,0 +1,212 @@
+package cash.z.ecc.android.ui.send
```
Comment (gmale) [inline app/src/main/java/cash/z/ecc/android/ui/send/SendAddressFragment.kt:1]: Why is this file added? Can we get away with not adding it (since it isn't used anymore)?

File: app/src/main/java/cash/z/ecc/android/ui/send/SendFragment.kt [hunk 3]
```diff
@@ -172,8 +186,8 @@ class SendFragment : BaseFragment<FragmentSendBinding>(),
     }
 
     /**
-    * To hide input Memo and reply-to option for T type address and show a info message about memo option availability
-    * */
+     * To hide input Memo and reply-to option for T type address and show a info message about memo option availability
+     * */
```
Comment (gmale) [inline app/src/main/java/cash/z/ecc/android/ui/send/SendFragment.kt:190]: minor: there seems to be an extra asterisk here that we can get rid of (This PR didn't add it but since the line is changing we might want to correct the formatting).
```suggestion
     */
```

**Review by gmale — COMMENTED** [review]
Looks good to me. Just had a few minor questions that don't necessarily block a merge.

**Comment by gmale** [comment]
Resolved merge conflicts from recently merged PR.

File: app/build.gradle (hunk 1 above)
Comment (nighthawk24) [reply app/build.gradle]: Reverted. I'll put up another PR to update the SDK version for the app & SDK.

File: app/src/main/java/cash/z/ecc/android/ui/send/SendAddressFragment.kt (hunk 2 above)
Comment (nighthawk24) [reply app/src/main/java/cash/z/ecc/android/ui/send/SendAddressFragment.kt]: Must be an artifact from Merge Conflict Resolution, removed.

File: app/src/main/java/cash/z/ecc/android/ui/send/SendFragment.kt (hunk 3 above)
Comment (nighthawk24) [reply app/src/main/java/cash/z/ecc/android/ui/send/SendFragment.kt]: Fixed.
